### ISSUE TITLE
Adding support for following the user's accent color

### ISSUE THREAD

**Issue by legofanlovessayori** [Maplespe/DWMBlurGlass#242]
**Describe the solution you'd like**
> A clear and concise description of what you want to happen.

Add an option that allows the user to use their Windows accent color for the glass color.

**Your reason (tell us why we need it)**
> Briefly describe your reasons for wanting to do this.

A lot of people (like myself) use this to make their Windows installs themed after Windows 7 or to add transparency back but its kinda painful to manually change the color every time when we change the wallpaper or use a different Windows accent color. 

**Describe alternatives you've considered**
> A clear and concise description of any alternative solutions or features you've considered.

We do currently have a fork of this that allows the use of the accent color (as mentioned in issue #234) but it would be nice to have said feature built in and not relying on forks.

**Comment by TheInsane101** [comment]
Duplicate of #86

**Comment by legofanlovessayori** [comment]
> Duplicate of #86

seems to be ignored tho

**Comment by xodus2002** [comment]
> Duplicate of #86 

It's technically not? That's just the color syncronization and using the colors from Windows color palette, instead of just manually add it. It's very frustrating to do that, like BigMuscle's AeroGlass has this function but not DWMBG.

**Comment by TheInsane101** [comment]
> > Duplicate of #86
> 
> seems to be ignored tho

The developers have seen it already and replied to it, saying it is low on their priority list. They have very clearly not ignored it; it is probably on their list of future plans somewhere.

**Comment by TheInsane101** [comment]
> > Duplicate of #86
> 
> It's technically not? That's just the color syncronization and using the colors from Windows color palette, instead of just manually add it. It's very frustrating to do that, like BigMuscle's AeroGlass has this function to use the colors from windows color palette but not DWMBG.

You're literally repeating what I and legofanlovessayori both said. How are they different?

**Comment by legofanlovessayori** [comment]
> > > Duplicate of #86
> > 
> > seems to be ignored tho
> 
> Read the replies: the developers have seen it already and replied to it, saying it is low on their priority list. They have very clearly not ignored it; it is probably on their list of future plans somewhere.

i mean, most users want that feature so its kinda pointless pushing it back

it just makes people fork it to add said feature

**Comment by TheInsane101** [comment]
That's a valid point, but nagging them about it is not going to change their plans. After all, they created this program and they can choose what happens to their vanilla fork. I have things I wish were included too, and all we can do is suggest them to them once, and then accept whatever decisions they make.

I am frustrated that the GUI is closed source, as it feels as if that is the thing that is handicapping all the forks from being as good as the original, but I can't expect the devs to do only what we want.

**Comment by legofanlovessayori** [comment]
> That's a valid point, but nagging them about it is not going to change their plans. After all, they created this program and they can choose what happens to their vanilla fork. I have things I wish were included too, and all we can do is suggest them to them once, and then accept whatever decisions they make.
> 
> I am frustrated that the GUI is closed source, as it feels as if that is the thing that is handicapping all the forks from being as good as the original, but I can't expect the devs to do only what we want.

do we even need the gui for the implementation
like the last 2 releases added features in irregardless of the gui

**Comment by TheInsane101** [comment]
> > That's a valid point, but nagging them about it is not going to change their plans. After all, they created this program and they can choose what happens to their vanilla fork. I have things I wish were included too, and all we can do is suggest them to them once, and then accept whatever decisions they make.
> > I am frustrated that the GUI is closed source, as it feels as if that is the thing that is handicapping all the forks from being as good as the original, but I can't expect the devs to do only what we want.
> 
> do we even need the gui for the implementation like the last 2 releases added features in irregardless of the gui

We absolutely do. It is primitive, clunky and slow to manually change the figures in hard-to-read text, save every change each time, and then uninstall and install the exe. Isn't it ironic that we're struggling to maintain an up to date GUI for a program that literally customises the appearance of an operating system's GUI?

**Comment by legofanlovessayori** [comment]
> > > That's a valid point, but nagging them about it is not going to change their plans. After all, they created this program and they can choose what happens to their vanilla fork. I have things I wish were included too, and all we can do is suggest them to them once, and then accept whatever decisions they make.
> > > I am frustrated that the GUI is closed source, as it feels as if that is the thing that is handicapping all the forks from being as good as the original, but I can't expect the devs to do only what we want.
> > 
> > do we even need the gui for the implementation like the last 2 releases added features in irregardless of the gui
> 
> We absolutely do. It is primitive, clunky and slow to manually change the figures in hard-to-read text, save every change each time, and then uninstall and install the exe. Isn't it ironic that we're struggling to maintain an up to date GUI for a program that literally customises the appearance of an operating system's GUI?

so you are telling me that we can't  implement the feature and then let the main dev that has access to whatever that gui framework is add the toggle after the fact

**Comment by Maplespe** [comment]
First, I feel I need to clarify a few points.
DWMBlurGlass does not rely on MiaoUI for any of its functions!
How do I change the configuration, you may ask? How do I refresh it?...
In fact if you read the source code a little bit you should be able to realize that
All the configuration modification code uses the public Windows API 'WritePrivateProfileStringW' and `GetPrivateProfileStringW`.
This is not private code, anyone can call it and modify it.

How do I refresh the configuration?
You simply send a message to DWM to refresh it
`PostMessageW(FindWindowW(L "Dwm", nullptr), WM_THEMECHANGED, 0, 0);`
The work is done by DWMBlurGlassExt and is completely independent of the GUI!

I use MiaoUI just for personal convenience. No one is stopping you from using other GUI libraries.

But what I'm saying is you guys can't take this as something the devs should be doing as a matter of course. **It's really very rude.**

I'm not saying you can't make requests and we'll take care of them, but if you really can't wait you should read the code yourselves, not keep rushing the developers and asking cheap questions.

**You should understand that all this work we are doing is unpaid, and I have my own school and work to complete, and can't be on this project every day to fulfill all of your requests.**

**Comment by Maplespe** [comment]
Finally to answer a few related questions.
I had planned to add features to follow accent in 2.1.0 including adding auto-injection and fixing the bug where bootstrap doesn't work.

But I've been so busy lately that I handed the project over to ALTaleX.
I didn't explain to him beforehand which features to update, but of course, this should have been done by me and I didn't ignore your requests.

I can't promise that I'll be able to update it lately, but and I'm disappointed by your attitudes and reactions.
And I'm certainly very grateful to everyone who has helped with this project,**but this is all unpaid work, and they should all be encouraged rather than criticized.**

**Comment by legofanlovessayori** [comment]
but that gui frontend is the issue


if we can't add stuff because its closed source then no one can help

**Comment by Maplespe** [comment]
> but that gui frontend is the issue
if we can't add stuff because its closed source then no one can help

As I said, you can write your own GUI read/write configs and refresh configs, there's no critical code that's closed source.
Just like those branch implementations.
The core code is in DwmBlurGlassExt.

**Comment by Maplespe** [comment]
@kfh83 That's okay. I appreciate your work. You've contributed a lot to this project.

**Comment by legofanlovessayori** [comment]
> > but that gui frontend is the issue
> if we can't add stuff because its closed source then no one can help
> 
> As I said, you can write your own GUI read/write configs and refresh configs, there's no critical code that's closed source.
> Just like those branch implementations.
> The core code is in DwmBlurGlassExt.

so what i get from this is

you want us to write our own gui because of the gui frontend you used is closed source

just rewrite it, theres literally a win32 frontend available
get in touch with those and just use a normal win32 app

**Comment by legofanlovessayori** [comment]
then at least provide a toggle to disable the custom assets (like most apps have)

that thing behaves the same as uwp apps on 3rd party msstyles with window borders and doesn't fit on anything that isn't 11

**Comment by Maplespe** [comment]
MEF uses WinUI3, which is a failure. The user needs to install 40MB of runtime libraries for this, when in reality it's only a few dozen KB of core code, the rest is GUI. it's hilarious.

**Comment by Maplespe** [comment]
Also, you may not be aware that WinUI3 has very many bugs and low performance. Lag and bugs are clearly visible in all programs that use WinUI3, such as Notepad, Task Manager, File Explorer (in win11)

**Comment by Maplespe** [comment]
I may provide a rudimentary win32ui in the future, but that's outside the scope of this discussion, and it's the same no matter who does it, it's not complicated.

**Comment by legofanlovessayori** [comment]
maybe uhh

contact the devs of that dog configurator app and use that?

**Comment by Endeade** [comment]
> Also, you may not be aware that WinUI3 has very many bugs and low performance. Lag and bugs are clearly visible in all programs that use WinUI3, such as Notepad, Task Manager, File Explorer (in win11)

Sorry if I may come across as rude, but those WinUI3 libraries are not _always_ laggy on _all hardware_. They are only laggy on low performance GPUs and it has been noted as that for like a bazillion times.

**Comment by Maplespe** [comment]
@Endeade 
These are manifested in every aspect, and you're only talking about some of them.
For example, some pages, navigation bar buttons switching animation cutoffs, and all sorts of other bugs. you even casually click a few more times in the Winui3 gallery he is likely to be buggy and crash. But in WinUI2 there are not so many problems.
I was also looking forward to winui3 at first I experienced it the first time it came out then, but it just turned out to be very disappointing.
Even now you can't deny that it relies on those messy runtime libraries.

BTW, I'm a C++ developer, and while I do know C# I don't use it often. So it doesn't work well for me.
I'd rather use Qt than WinUI.

**Comment by Maplespe** [comment]
> Also i should probably mention in this issue: I've already tried reading the AccentColor key in HKCU\Software\Microsoft\Windows\DWM:

> And as you can see, it is using the Accent color set in settings. However...

> It would remain that color no matter what you did except log out of your windows session and log back in (or if you restarted your computer). Not even reloading DWMBlurGlass would get it to change. What perplexes me is that i wrote a simple C# program that simply printed the value of that Registry key to CMD, and in fact when i would change it, it would reflect in the C# program. I'm not sure what was going on in my modded DWMBg.

Maybe you need to update the value in the g_config variable and refresh it.
You can try to send a message or refresh all Backdrops directly using the Visual Manager class in the code.

But that's ok, ALTaleX531 will do all that soon.

**Comment by Endeade** [comment]
@Maplespe 
I understand your worries regarding WinUI 3, but I have used multiple low end and high end systems with Windows 11 and I have never seen any issues with WinUI 3.
I am in no way looking to force you to become a C# developer, god forbid, but I believe you got my message wrong. As many users of DWMBG stated by now, why not implement following the user's color palette firstly in config-only files to test it out and then you can look what you can do about the UI, I mean why not? I mean, heck, even I when I used to develop apps and when I had a bucket list for features and I saw that people requested it, I didn't put it at the bottom of the list, but I pushed it to the top!
Anyway, I guess we'll see about this later.

**Comment by Maplespe** [comment]
@Endeade  As I think I explained above, I planned to add these features in version 2.1.0 and I didn't ignore the requests or prioritize them last. However, the project is now handed over to ALTaleX531, and I didn't tell him ahead of time which features had to be implemented.
So these features should be added in the next version.

**Comment by Endeade** [comment]
Well, then I guess we're done here!
Cheers!

**Comment by TheInsane101** [comment]
@Maplespe I want to apologise for my rude attitude towards the GUI thing. I used my frustration about the GUI as an example of something that I want changed that I accept is the developers' decision and that I'm not entitled to it at the end of the day, but I could have been much more polite about it and I realise I crossed the line.

@legofanlovessayori I want to apologise for being rude to you too. I could have explained my points better without being snappy. I see you have been acting entitled and rude yourself, but that did not justify me setting a terrible example myself.
